src/Razor/src/Microsoft.AspNetCore.Razor.Language/DefaultRazorIntermediateNodeLoweringPhase.cs

```diff
@@ -1885,36 +1885,98 @@ public override void VisitMarkupElement(MarkupElementSyntax node)
             {
                 _document.Diagnostics.Add(
                     ComponentDiagnosticFactory.Create_UnsupportedComponentImportContent(BuildSourceSpanFromNode(node)));
+
+                base.VisitMarkupElement(node);
             }
 
             public override void VisitMarkupCommentBlock(MarkupCommentBlockSyntax node)
             {
                 _document.Diagnostics.Add(
                     ComponentDiagnosticFactory.Create_UnsupportedComponentImportContent(BuildSourceSpanFromNode(node)));
+
+                base.VisitMarkupCommentBlock(node);
             }
 
             public override void VisitMarkupTagHelperElement(MarkupTagHelperElementSyntax node)
             {
                 _document.Diagnostics.Add(
                     ComponentDiagnosticFactory.Create_UnsupportedComponentImportContent(BuildSourceSpanFromNode(node)));
+
+                base.VisitMarkupTagHelperElement(node);
             }
 
-            public override void VisitCSharpImplicitExpression(CSharpImplicitExpressionSyntax node)
+            public override void VisitCSharpExplicitExpression(CSharpExplicitExpressionSyntax node)
             {
                 _document.Diagnostics.Add(
                     ComponentDiagnosticFactory.Create_UnsupportedComponentImportContent(BuildSourceSpanFromNode(node)));
+
+                base.VisitCSharpExplicitExpression(node);
             }
 
-            public override void VisitCSharpExplicitExpression(CSharpExplicitExpressionSyntax node)
+            public override void VisitCSharpImplicitExpression(CSharpImplicitExpressionSyntax node)
             {
+                if (node.FirstAncestorOrSelf<SyntaxNode>(n => n is MarkupStartTagSyntax || n is MarkupEndTagSyntax) != null)
```

Comment:
Add a comment stating why we care about implicit expressions in _Imports.razor for any future readers.

Reply:
Added